src/cart/crt_hg.c

```diff
@@ -1298,6 +1298,7 @@ crt_hg_reply_error_send(struct crt_rpc_priv *rpc_priv, int error_code)
 	D_ASSERT(error_code != 0);
 
 	hg_out_struct = &rpc_priv->crp_pub.cr_output;
+	rpc_priv->crp_reply_pending = 0;
```

Comment:
We should probably only clear it if HG_Respond succeeded and keep reply pending if it failed

Reply:
If it fails what would the code path be? Would it retry again and would the flag being cleared create a problem?

Reply:
It should retry again then before rpc is destroyed (which is where we previously were hitting second reply case)

Reply:
ok resubmitted